### PR TITLE
fix: propagating mesh annotations to transient pods

### PR DESCRIPTION
**Pull request by bc185174**
**What this PR does / why we need it**:

Mesh annotations are not replicating to pods set by the DV/PVCs. Allows users to unmesh these pods by default.

- Prep claim pod.
- Size detection pod.
- Importer pod.
- Clone source pod.
- Upload pod.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Mesh annotations now copied from the DV/PVCs to the following pods.

- Prep claim pod.
- Size detection pod.
- Importer pod.
- Clone source pod.
- Upload pod.
```

### CODE REVIEW

**Comment by mhenriks** [comment]
Any disadvantage to adding these annotations all the time?  Could they ever be undesirable?  Could they ever somehow conflict?  If so me may want to add a param to CDI when we want annotations.  If the answer to both questions is no then I guess this approach fine

**Comment by mhenriks** [comment]
> Not really no - by default they are disabled and only overwritten by the end users.

Sorry but I'm not quite parsing this

What do you mean by "disabled"?  Obviously if there is no service mesh the annotations will have no effect.  Is that what you mean?

Also confused about "overwritten".  When and why?

**Comment by bc185174** [comment]
> > Not really no - by default they are disabled and only overwritten by the end users.
> 
> Sorry but I'm not quite parsing this
> 
> What do you mean by "disabled"? Obviously if there is no service mesh the annotations will have no effect. Is that what you mean?
> 
> Also confused about "overwritten". When and why?

Apologies - by default those annotations are set to `false` (istio) or `disabled` (linkerd) regardless of whether user has added them or not. They'll only change if a user applies the annotation, essentially patching and overwriting the default value.

File: pkg/controller/clone-controller.go [hunk 1]
```diff
@@ -548,8 +548,10 @@ func MakeCloneSourcePodSpec(sourceVolumeMode corev1.PersistentVolumeMode, image,
 			Name:      cloneSourcePodName,
 			Namespace: sourcePvcNamespace,
 			Annotations: map[string]string{
-				cc.AnnCreatedBy: "yes",
-				AnnOwnerRef:     ownerRefAnno,
+				cc.AnnCreatedBy:                  "yes",
+				AnnOwnerRef:                      ownerRefAnno,
+				cc.AnnPodSidecarInjectionIstio:   sourcePvc.Annotations[cc.AnnPodSidecarInjectionIstio],
```
Comment (akalenyu) [inline pkg/controller/clone-controller.go:553]: You should be able to use `SetPvcAllowedAnnotations` for passing annotations from the data volume to the transfer pods

Comment (bc185174) [reply pkg/controller/clone-controller.go]: Yeah - both prep-claim and the size detection pod were missing this.

Comment (akalenyu) [reply pkg/controller/clone-controller.go]: Yeah looks much better now

File: pkg/controller/datavolume/pvc-clone-controller.go [hunk 2]
```diff
@@ -677,13 +677,14 @@ func (r *PvcCloneReconciler) makeSizeDetectionPodSpec(
 		pod.Annotations[cc.AnnOwnerUID] = string(dv.UID)
 	}
 
+	cc.SetPvcAllowedAnnotations(pod, sourcePvc)
```
Comment (akalenyu) [inline pkg/controller/datavolume/pvc-clone-controller.go:680]: I think you want targetPvc/targetDv instead of source here

**Comment by akalenyu** [comment]
/hold
small issue
https://github.com/kubevirt/containerized-data-importer/pull/3186/files#r1565763489

**Pull request merged**